src/main/java/app/bpartners/api/endpoint/event/S3Conf.java

```diff
@@ -21,32 +21,42 @@ public class S3Conf {
   private final String s3Endpoint;
   private final String bucketName;
   private final Region region;
+  private final AwsConf awsConf;
 
   private final String env;
 
   public S3Conf(
       @Value("${aws.bucket.name}") String bucketName,
       @Value("${env}") String env,
       @Value("${aws.endpoint}") String s3Endpoint,
-      @Value("eu-west-3") Region region) {
+      @Value("eu-west-3") Region region,
+      AwsConf awsConf) {
     this.bucketName = bucketName;
     this.env = env;
     this.s3Endpoint = s3Endpoint;
     this.region = region;
+    this.awsConf = awsConf;
   }
 
   @Bean
   public S3Client getS3Client() {
     try {
-      return S3Client.builder().endpointOverride(new URI(s3Endpoint)).region(region).build();
+      return S3Client.builder()
+          .credentialsProvider(awsConf.awsCredentialsProvider())
```

Comment:
also this is not how you use a bean, you'd rather inject the Bean instead of injecting awsConf